### DISCUSSION TITLE
Unique Identifier

### DISCUSSION THREAD

**Discussion by SergeantPanda** [arabcoders/watchstate#327] [General] [answered]
I'm having issues getting Plex fully working. I have the scheduled sync working (I believe) but the webhook is failing.

I am getting this error message after selecting the user: 
`Unable to get [MYUSERNAME] access token. rerun the command with [-vvv --context --trace] flags for more info or check logs.`

I suspect it's because I'm not entering the correct Unique Identifier but I'm not sure where I'm supposed to get that from. I have tried everything I can think of so far with no luck.

Any help would be greatly appreciated.

Thank you!

**Comment by arabcoders** [comment]
@SergeantPanda Could you please reurn the same command with the listed flags to have more info regarding the error. i suspect the server has limited token thus cant generate new ones.

**Reply by SergeantPanda** [reply]
Definitely!

Here is the output of that, I just removed my email and thumbs.

What is it looking for exactly for the unique ID though? Can it be anything I want or is it something specific?

```
Attempting to get users list from plex.tv API. Please wait...
[2023-08-18T13:38:13-05:00] DEBUG: Requesting [plex] Users list. { (url: https://plex.tv/api/v2/home/users/) }
[2023-08-18T13:38:14-05:00] DEBUG: Parsing [plex] user list payload. { (url: https://plex.tv/api/v2/home/users/), (trace: [ (id: 89550), (name: SergeantPanda's home), (guestUserID: 5104320), (guestUserUUID: c949056bc6839bb9), (guestEnabled: true), (subscription: true), (users: [ (0: [ (id: 1223995), (uuid: 7b529d25c1f43b7f), (title: SergeantPanda), (username: SergeantPanda), (email: myemail@gmail.com), (friendlyName: None), (thumb: ), (hasPassword: true), (restricted: false), (updatedAt: 1430446565), (restrictionProfile: None), (admin: true), (guest: false), (protected: false) ]), (1: [ (id: 5104320), (uuid: c949056bc6839bb9), (title: Guest), (username: None), (email: None), (friendlyName: Guest), (thumb: ), (hasPassword: true), (restricted: true), (updatedAt: 1431466531), (restrictionProfile: None), (admin: false), (guest: true), (protected: false) ]), (2: [ (id: 5004539), (uuid: 01033fe64bc317e3), (title: Kiddos), (username: None), (email: None), (friendlyName: Kiddos), (thumb: ), (hasPassword: true), (restricted: true), (updatedAt: 1691596948), (restrictionProfile: teen), (admin: false), (guest: false), (protected: false) ]) ]) ]) }
Select [plex] User. [Default: SergeantPanda]
------------------
> 
  [0] SergeantPanda
  [1] Guest
  [2] Kiddos
 > 
Unable to get [SergeantPanda] access token. rerun the command with [-vvv --context --trace] flags for more info or check logs.
```

**Reply by arabcoders** [reply]
The unique id is a random string generated on the server side, plex in their infinite wisdom triggers webhooks for all servers even if you didn't add it to it, thus we need to differentiate which server is sending the webhook events. thus the unique identifier. 

Make sure the user in question actually has access to the server that is associated in config. Please take look  at this issue #325

**Reply by arabcoders** [reply]
You can also try this command `plex:accesstoken plex 7b529d25c1f43b7f -vvv --trace --context` which should give more info are you the owner of the server?

**Reply by SergeantPanda** [reply]
Still struggling with this. I switched to the dev release just in case and am still getting the same issue.

I am definitely the server owner. Still not sure I'm using the correct unique ID, where exactly does one find what it's looking for? The one I pulled is from the config file.

Here is the output of the command (again, scrubbed some stuff out):

```
-snipped- (please check your logs for tokens)
[2023-08-18T13:55:44-05:00] ERROR: Response had [8] of associated servers non match [plex] unique identifier. { (backend_id: 7b529d25c1f43b7f) }
[2023-08-18T13:55:44-05:00] ERROR: No permanent access token was found for [plex_user] in [plex] response. Likely invalid unique identifier was selected or plex.tv API error, check https://status.plex.tv or try running same command with [-vvv --trace --context] flags for more information. { (user_id: 7b529d25c1f43b7f) }

In PlexClient.php line 362:
                                                                                                                                                                             
  [RuntimeException]                                                                                                                                                         
  No permanent access token was found for [plex_user] in [plex] response. Likely invalid unique identifier was selected or plex.tv API error, check https://status.plex.tv   
  or try running same command with [-vvv --trace --context] flags for more information.                                                                                      
                                                                                                                                                                             

Exception trace:
  at /opt/app/src/Backends/Plex/PlexClient.php:362
 App\Backends\Plex\PlexClient->getUserToken() at /opt/app/src/Backends/Plex/Commands/AccessTokenCommand.php:104
 App\Backends\Plex\Commands\AccessTokenCommand->runCommand() at /opt/app/src/Command.php:47
 App\Command->execute() at /opt/app/vendor/symfony/console/Command/Command.php:326
 Symfony\Component\Console\Command\Command->run() at /opt/app/vendor/symfony/console/Application.php:1063
 Symfony\Component\Console\Application->doRunCommand() at /opt/app/vendor/symfony/console/Application.php:320
 Symfony\Component\Console\Application->doRun() at /opt/app/vendor/symfony/console/Application.php:174
 Symfony\Component\Console\Application->run() at /opt/app/src/Libs/Initializer.php:140
 App\Libs\Initializer->console() at /opt/app/bin/console:63
```

**Reply by arabcoders** [reply, marked as answer]
I mean it's showing exactly what is expected ` ERROR: Response had [8] of associated servers non match [plex] unique identifier.` could you remove the config re-add the server and check the unique identifier? you shouldn't have to manually enter it unless your access token is limited one. If this also doesn't work please jump into discord and let's have talk about it to diagnose it further and faster. 

i parsed the logs you have sent and noticed that you set a user id `7b529d25c1f43b7f` as the unique identifier which is incorrect. Please reset your config and start new do not manually enter the unique identifier. let the one that is set as defualt to be used. the unique identifier is something like `9350663c708f6c78f613e82d3.......` which you can see in the logs

**Reply by SergeantPanda** [reply]
Ok perfect, thank you for the clarification! I had started with that initially but had the same problem. I just went ahead and recreated it and still had the same issue.

At least now I can confirm that the Unique ID that WatchState is looking for is what Plex calls the "ProcessedMachineIdentifier" which can be found in the Preferences.xml file in the plexmediaserver folder.

I will hop into the Discord.

Thank you very much for your help so far!

**Reply by arabcoders** [reply]
No worries, we the unique identifier has many names in plex world sadly, ClientIdentifier, ProcessedMachineIdentifier and unique id any probably more =)
